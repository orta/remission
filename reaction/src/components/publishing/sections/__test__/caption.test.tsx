import * as React from "react"
import * as renderer from "react-test-renderer"

import "jest-styled-components"

import Caption from "../caption"

it("renders a saved caption properly", () => {
  const embed = renderer.create(<Caption caption="<p>This is a saved caption</p>" />).toJSON()
  expect(embed).toMatchSnapshot()
})

it("renders a child even if a caption is provided", () => {
  const embed = renderer
    .create(<Caption caption="<p>This is a saved caption</p>"><div>Here is a passed child</div></Caption>)
    .toJSON()
  expect(embed).toMatchSnapshot()
})
