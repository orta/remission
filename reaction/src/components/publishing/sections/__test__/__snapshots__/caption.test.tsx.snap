// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a child even if a caption is provided 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
  margin: 10px 0 10px 0;
}

.c1 {
  padding: 0;
}

.c1 > p,
.c1 p,
.c1 .public-DraftEditorPlaceholder-root {
  font-family: Unica77LLWebRegular ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 14px;
  line-height: 1.4em;
  color: #999;
  margin: 0;
}

.c1 > a,
.c1 a {
  color: #999;
}

.c1 > a:hover,
.c1 a:hover {
  color: black;
}

@media (max-width: 600px) {
  .c0 {
    padding: 0px 10px;
  }
}

@media (max-width: 600px) {
  .c1 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div>
      Here is a passed child
    </div>
  </div>
</div>
`;

exports[`renders a saved caption properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-between;
  -webkit-justify-content: space-between;
  -ms-flex-pack: space-between;
  justify-content: space-between;
  margin: 10px 0 10px 0;
}

.c1 {
  padding: 0;
}

.c1 > p,
.c1 p,
.c1 .public-DraftEditorPlaceholder-root {
  font-family: Unica77LLWebRegular ,      'Arial',      'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 14px;
  line-height: 1.4em;
  color: #999;
  margin: 0;
}

.c1 > a,
.c1 a {
  color: #999;
}

.c1 > a:hover,
.c1 a:hover {
  color: black;
}

@media (max-width: 600px) {
  .c0 {
    padding: 0px 10px;
  }
}

@media (max-width: 600px) {
  .c1 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>This is a saved caption</p>",
        }
      }
    />
  </div>
</div>
`;
